src/coreclr/tools/aot/ILCompiler/Program.cs

```diff
@@ -74,118 +74,9 @@ public int Run()
-                optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("aes");
-                optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("pclmul");
-                optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("movbe");
-                optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("popcnt");
-                optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("lzcnt");
-
-                // If AVX was enabled, we can opportunistically enable instruction sets which use the VEX encodings
-                Debug.Assert(InstructionSet.X64_AVX == InstructionSet.X86_AVX);
-                if (supportedInstructionSet.HasInstructionSet(InstructionSet.X64_AVX))
-                {
-                    optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("fma");
-                    optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("bmi");
-                    optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("bmi2");
-                    optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("avxvnni");
-                }
-            }
-            else if (targetArchitecture == TargetArchitecture.ARM64)
-            {
-                optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("aes");
-                optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("crc");
-                optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("sha1");
-                optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("sha2");
-                optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("lse");
-                optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("rcpc");
-            }
-
-            optimisticInstructionSetSupportBuilder.ComputeInstructionSetFlags(out var optimisticInstructionSet, out _,
-                (string specifiedInstructionSet, string impliedInstructionSet) => throw new NotSupportedException());
-            optimisticInstructionSet.Remove(unsupportedInstructionSet);
-            optimisticInstructionSet.Add(supportedInstructionSet);
-
-            var instructionSetSupport = new InstructionSetSupport(supportedInstructionSet,
-                                                                    unsupportedInstructionSet,
-                                                                    optimisticInstructionSet,
-                                                                    InstructionSetSupportBuilder.GetNonSpecifiableInstructionSetsForArch(targetArchitecture),
-                                                                    targetArchitecture);
+            InstructionSetSupport instructionSetSupport = Helpers.ConfigureInstructionSetSupport(Get(_command.InstructionSet), skipRcpc: false, targetArchitecture, targetOS,
```

Comment:
Why do we need the `skipRcpc` flag? Is the difference between r2r and native aot intentional?

Reply:
@EgorBo explained that rcpc makes it less efficient than the fallback: https://github.com/dotnet/runtime/pull/76537#discussion_r986369688.

I just realized that it was added to ILC for osx-arm64 in #67384,  which was months before we were able to even run the first nativeaot hello-world program on the platform; do we not have the efficiency concern there?

Reply:
@am11 I think we should just drop `rcpc` from being an "optimistic" instruction from both R2R and ILS so there no need for skipRcpc here.

It supposed to slightly improve perf for volatile reads, basically replace `ldar` with `ldapr` so we don't want to have here a runtime check in case if it's not supported.

Reply:
Also, I do not think that the JIT is actually going to generate the optimistic conditions for `rcpc` in native aot today. It should be ok to drop this - it does not actually change anything.